password_security/models/res_users.py

```diff
@@ -81,6 +81,8 @@ def check_password(self, password):
             password_regex.append(r'(?=.*?\W)')
         password_regex.append('.{%d,}$' % company_id.password_length)
         if not re.search(''.join(password_regex), password):
+            self.env.cr.rollback()
```

Comment:
Yeah, I find the real issue.
The method `def _set_password(` don't run a rollback if an exception is raised.
So I just changed the inherited method to write validation
I removed rollback and clear caches